e2e/framework/machine.go

```diff
@@ -54,3 +58,52 @@ func FilterRunningMachines(machines []*clusterv1.Machine) []*clusterv1.Machine {
 
 	return result
 }
+
+// GetMachine get a machine by its name.
+func GetMachine(cl client.Client, name string, namespace string) (*clusterv1.Machine, error) {
+	machine := &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	Eventually(komega.Get(machine), time.Minute, RetryShort).Should(SatisfyAny(
+		Succeed(),
+		WithTransform(apierrors.IsNotFound, BeTrue()),
```

Comment:
If we just created a machine, we wouldn't want to have it return `NotFound` right? What's the use case for this util if it will return an empty Machine when the Machine isn't found?